ddtrace/ext/ci.py

```diff
@@ -295,14 +304,29 @@ def extract_circle_ci(env):
         JOB_NAME: env.get("CIRCLE_JOB"),
         PROVIDER_NAME: "circleci",
         WORKSPACE_PATH: env.get("CIRCLE_WORKING_DIRECTORY"),
-        # OrderedDict is necessary for comparing against a fixture in testing
         _CI_ENV_VARS: json.dumps(
-            OrderedDict(
-                [
-                    ("CIRCLE_WORKFLOW_ID", env.get("CIRCLE_WORKFLOW_ID")),
-                    ("CIRCLE_BUILD_NUM", env.get("CIRCLE_BUILD_NUM")),
-                ]
-            ),
+            {
+                "CIRCLE_WORKFLOW_ID": env.get("CIRCLE_WORKFLOW_ID"),
+                "CIRCLE_BUILD_NUM": env.get("CIRCLE_BUILD_NUM"),
+            },
+            separators=(",", ":"),
+        ),
+    }
+
+
+def extract_codefresh(env):
+    # type: (MutableMapping[str, str]) -> Dict[str, Optional[str]]
+    """Extract CI tags from Codefresh environ."""
+    return {
+        git.BRANCH: env.get("CF_BRANCH"),
+        PIPELINE_ID: env.get("CF_BUILD_ID"),
+        PIPELINE_NAME: env.get("CF_PIPELINE_NAME"),
+        PIPELINE_NUMBER: env.get("CIRCLE_BUILD_NUM"),
+        PIPELINE_URL: "https://g.codefresh.io/build/{0}".format(env.get("CF_BUILD_ID")),
+        JOB_NAME: env.get("CF_STEP_NAME"),
+        PROVIDER_NAME: "codefresh",
+        _CI_ENV_VARS: json.dumps(
+            {"CF_BUILD_ID": env.get("CF_BUILD_ID")},
```

Comment:
Since you're calling several times `env.get("CF_BUILD_ID")`, why not store it's value to a variable for improved reuse?

Reply:
Great suggestion, added!